ecl/hql/hqldesc.cpp

```diff
@@ -113,15 +129,98 @@ static void expandRecordSymbolsMeta(IPropertyTree * metaTree, IHqlExpression * r
     }
 }
 
+void expandScopeMeta(IPropertyTree * meta, IHqlExpression * expr) {
+    if (expr->hasAttribute(virtualAtom))
+        meta->setPropBool("@virtual", true);
+
+    if (expr->hasAttribute(interfaceAtom)){
```

Comment:
Roger and I decided to put the development of type system once we have completed first iteration of ecldoc. Since Types can have complicated structures (for eg DATASET(rec_1), SET of STRING, etc), it will difficult to represent the relationships between different parts within the type correctly in an attribute.